src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -173,15 +179,159 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
+                                        async () => change = await client.GetKeyValueChange(watchedKv, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                                }
+                                else
+                                {
+                                    // Load the key-value in case the previous load attempts had failed
+
+                                    try
+                                    {
+                                        await CallWithRequestTracing(
+                                            async () => watchedKv = await client.GetConfigurationSettingAsync(watchedKey, watchedLabel, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                                    }
+                                    catch (RequestFailedException e) when (e.Status == (int)HttpStatusCode.NotFound)
+                                    {
+                                        watchedKv = null;
+                                    }
+
+                                    if (watchedKv != null)
+                                    {
+                                        change = new KeyValueChange()
+                                        {
+                                            Key = watchedKv.Key,
+                                            Label = watchedKv.Label.NormalizeNull(),
+                                            Current = watchedKv,
+                                            ChangeType = KeyValueChangeType.Modified
+                                        };
+                                    }
+                                }
+
+                                // Check if a change has been detected in the key-value registered for refresh
+                                if (change.ChangeType != KeyValueChangeType.None)
+                                {
+                                    keyValueChanges[changeWatcher] = change;
+
+                                    if (changeWatcher.RefreshAll)
+                                    {
+                                        refreshAll = true;
+
+                                        break;
+                                    }
+                                }
+                            }
+
+                            if (refreshAll)
+                            {
+                                // Trigger a single refresh-all operation if a change was detected in one or more key-values with refreshAll: true
+                                applicationSettings = await LoadAll(client, cancellationToken).ConfigureAwait(false);
+
+                                return;
+                            }
+
+                            changedKeyValuesCollection = await GetRefreshedKeyValueCollections(client, cancellationToken).ConfigureAwait(false);
+
+                            return;

```

Comment:
This `return` is at the end of func. Is it necessary?

Reply:
It's not necessary. It's a stray return from the version where it was initially returning a value, I made a change to just return but overlooked cleaning up `return` statement that was then unnecessary.